### ISSUE TITLE
S1542 FP: Don't raise for overridden members and interface implementations

### ISSUE THREAD

**Issue by PavlinII** [SonarSource/sonar-dotnet#4945]
### Description

S1542 Rename function 'xxx' to match the regular expression: 'xxx' should not raise for overriden members and interface implementations

### Repro steps

```VB.NET
Public MustInherit Class Base

    Public MustOverride Function rqSomething() As Object ' Noncompliant

End Class

Public Interface ISomething

    Function rqFirst() As Object      ' Noncompliant
    Function rqSecond() As Object     ' Noncompliant

End Interface

Public Class Sample
    Inherits Base
    Implements ISomething

    Public Overrides Function rqSomething() As Object   ' False Positive, can't change
        Throw New NotImplementedException()
    End Function

    Public Function rqFirst() As Object Implements ISomething.rqFirst   ' False Positive, should not raise
        Throw New NotImplementedException()
    End Function

    Public Function rqRenamed() As Object Implements ISomething.rqSecond ' Noncompliant, because existing name doesn't match the interface member
        Throw New NotImplementedException()
    End Function
End Class
```
